neso_solar_consumer/script/fetch_nl_data.py

```diff
@@ -0,0 +1,162 @@
+import os
```

Comment:
I think we perhaps need to seperate 
1. a script like it is / was, which can go in the top level folder upder scripts.
2. a function that pulls the latest solar data. Lots of values here should have defaults and no `args` functions